src/common/darktable.c

```diff
@@ -1073,17 +1089,22 @@ int dt_init(int argc, char *argv[], const gboolean init_gui, const gboolean load
   //  2 mipmap size
   //  3 opencl available
   static int fractions[20] = {
-      512,   32, 128, 600,  // default
+      512,   32, 128, 700,  // default
```

Comment:
This has nothing to do with the actual fix, just a simple usability suggestion: could those ratios be percentages, which people are more familiar with (especially now, that the setting is exposed to users via `darktablerc`)? I understand that `/ 100` is more expensive than `/ 1024`, but we won't be calculating that ratio a million times.
Another usability suggestion: log the calculated values. It would make diagnostics easier.

Reply:
I chose 1024 to stay with kb

Reply:
Then maybe I misunderstood the calculation. To me, expressing the desired ratio as:
`total_mem / 1024lu * fraction`
reads: 'for each 1 kB of memory, use `fraction` bytes'.

`total_mem / 100lu * percentage` (or even `total_mem *percentage / 100lu`) seems more natural (I don't think we have to worry about overflows with `unsigned long`).